### ISSUE TITLE
"Use of future reserved word in strict mode" false positive with eslint-env es6

### ISSUE THREAD

**Issue by cvrebert** [eslint/eslint#2134]
ESLint version: `0.17.1` (and 67936e0d571c613ac54f4097b4235e6411af0b79 gives the same results)

`foo.js`:
```js
/*eslint-env es6 */
let x = 42;
```

`.eslintrc`:
```json
{
    "ecmaFeatures": { "modules": true }
}
```

The false positive:
```
$ ./node_modules/.bin/eslint foo.js 

foo.js
  2:1  error  Use of future reserved word in strict mode

✖ 1 problem (1 error, 0 warnings)
```

**Comment by nzakas** [comment]
Most likely we're not apply `ecmaFeatures` for inline environments correctly. As a workaround, you can try adding `env: { es6: true }` to your config file.

**Comment by satya164** [comment]
Facing the same issue. It's not optimal to have it in the config file in an ES5 and ES6 mixed environment.

**Comment by nzakas** [comment]
You can toggle it on and off in different config files `es6: true` in one and `es6: false` in another.

**Comment by satya164** [comment]
If I use 2 different config files, won't I have to duplicate all the configs, and they'll get out of sync?

I use the extension `.es6` for my ES6 files. May be there could be config option to enable certain options for certain file extensions? May be I'm thinking too far.

**Comment by ilyavolodin** [comment]
Ok, I started working on this (I think it's actually an easy fix), but run into a bunch of things that I can't test and explain. Here's the code that currently applies inline environments: https://github.com/eslint/eslint/blob/master/lib/eslint.js#L306
From what I can tell, that code is broken and currently doesn't do anything at all for rules (verified) as it tries to merge in `true` value into existing object.
This is the only code I can find that deals with `eslint-env` comment. However, somehow, global variables are being set correctly when enabling environments that way. I can't find that code anywhere.
I also don't see any way to verify that after I set `eslint-env` rules will be enabled in config and ecmaFeatures will get copied. eslint doesn't expose any of those options unfortunately. So I don't think it's possible to unittest my fix.

**Comment by satya164** [comment]
I tested the fix and it seems to be working. Hope to see it in the stable version soon.

Thanks a lot \o/

**Comment by nzakas** [comment]
Closed by mistake. 

We still have an issue where ecmaFeatures aren't applied because we don't read comments until after parsing has already happened.

**Comment by ilyavolodin** [comment]
I'm not really sure if it's worth keeping this issue open. I don't really see a way to fix this without two-stage parsing.

**Comment by satya164** [comment]
It's such a basic functionality. It needs fixing.

**Comment by nzakas** [comment]
@ilyavolodin we can't leave this. Either we remove the es6 environment or we figure out a way to make this work. 

We could, in theory, just search for `/*eslint-env` and see if there's `es6:true` before parsing.
